### ISSUE TITLE
Strict mode: hash is required to be imported, but glint reports it as not used.

### ISSUE THREAD

**Issue by NullVoxPopuli** [typed-ember/glint#374]
Sample gjs:
```js
import { hash } from '@ember/helper';

<template>{{yield (hash x=1)}}</template>
```

Sample Error:
```
app/components/limber/output/compiler/index.gts:3:1 - error TS6133: 'hash' is declared but its value is never read.

3 import { hash } from '@ember/helper';
```

Since removing the import creates a runtime error, "hash is not defined", I think Glint should not report the hash import as never read?

I'm using 
 - ember-source 4.5
 - ember-template-imports 3.0.1
 - @glint* 0.9.1

**Comment by NullVoxPopuli** [comment]
you and me both <3

**Comment by dfreeman** [comment]
I remember during discussions around the default helper manager, there was some discussion that it would be easier to avoid supporting named args if there were dedicated object syntax (basically just 1:1 parity with JS function invocation), but I don't know that that was ever actually captured anywhere.

**Comment by ef4** [comment]
Instead of compiling `(hash a=b)` directly to `({ a: b })` it could compile to `(χ.noop(hash), { a: b })`.

I confirmed that this keeps typescript's strict object literal handling, in the sense that it makes unused keys into type errors.

**Comment by dfreeman** [comment]
Ah, that's clever. Using an empty array literal as a canary (both because I know that's particularly fragile inference behavior in TS and because it could affect us with `{{array}}`), it looks like that approach _can_ impact the propagation of type info: https://www.typescriptlang.org/play?#code/DYUwLgBAHhC8EG0C6BuAUFFED02IEMA7AT2TQwDoAHAVwGcALACgEYAmAZgEp1Mc9CNALYAjEACcy5NKEjE4EJgAYANIiQ80xLLgiEQANwlTi1es3bcdeAILiA5sJCFIAewBmEMMSogIAckFRCX8IAEs6PVdIfDo6MPtCfBFQL1cIKnxxfCFwCQgPLx8-QMMQiiA

I think we'd want to be careful to make sure that none of those are situations we actually care about in a template context, but if not I like the comma expression approach. Thanks for the suggestion!

**Comment by NullVoxPopuli** [comment]
with `hash`, don't the individual values track separately as well? like, if we switched to an object literal, we'd forcibly entangle all values with the whole object?

**Comment by ef4** [comment]
It already compiles to an object literal. But this isn't code that runs, it's just code that tsc looks at.

**Comment by wagenet** [comment]
Is this just waiting on someone to do the work?
